app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -534,23 +534,25 @@ private void VibrateNotifyMakeNoise(Context context, AlertType alert, String bgV
                 .setDeleteIntent(snoozeIntent(context, minsFromStartPlaying));
 
         if (profile != ALERT_PROFILE_VIBRATE_ONLY && profile != ALERT_PROFILE_SILENT) {
-            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES) {
-                // Before this, we only vibrate...
-                float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
-                volumeFrac = Math.min(volumeFrac, 1);
-                if (profile == ALERT_PROFILE_MEDIUM) {
-                    volumeFrac = (float) 0.7;
-                }
-                Log.d(TAG, "VibrateNotifyMakeNoise volumeFrac = " + volumeFrac);
-                boolean overrideSilent = alert.override_silent_mode;
-                boolean forceSpeaker = alert.force_speaker;
-                if (notSilencedDueToCall()) {
-                    if (overrideSilent || isLoudPhone(context)) {
-                        playFile(context, alert.mp3_file, volumeFrac, forceSpeaker, overrideSilent);
-                    }
-                } else {
-                    Log.i(TAG, "Silenced Alert Noise due to ongoing call");
+            float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
+            // While minsFromStartPlaying <= MAX_VIBRATING_MINUTES, we only vibrate ...
+            volumeFrac = Math.max(volumeFrac, 0); // Limit volumeFrac to values greater than and equal to 0
+            volumeFrac = Math.min(volumeFrac, 1); // Limit volumeFrac to values less than and equal to 1
```

Comment:
I believe we should reevaluate the complete formula because currently, this line always returns 0 for minsFromStartPlaying ∈ [0..2] and always returns 1 for minsFromStartPlaying ∈ [0..2].

| `minsFromStartPlaying` | `volumeFrac` | `Math.min(Math.max(volumeFrac, 0), 1)` |
|:----------------------:|-------------:|---------------------------------------:|
|            0           |         -2.0 |                                    0.0 |
|            1           |         -1.0 |                                    0.0 |
|            2           |          0.0 |                                    0.0 |
|            3           |          1.0 |                                    1.0 |
|            4           |          2.0 |                                    1.0 |
|            5           |          3.0 |                                    1.0 |

Reply:
I have tested the PR and it works.  I have been using it since I made the last commit change.

I suggest you test my PR as is and if you find a problem, please let me know.

Reply:
The volume behaves exactly as it did before except, with the new setting disabled, it starts from 30% instead 0f 0.  That's the only change.
So, it is incorrect to say that there is no ascending volume.